src/main/java/yoda/commands/DeadlineCommand.java

```diff
@@ -0,0 +1,67 @@
+package yoda.commands;
+
+import yoda.TaskList;
+import yoda.exceptions.InvalidInputException;
+import yoda.tasks.Deadline;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a command that adds a deadline task to the task list.
+ */
+public class DeadlineCommand extends Command {
+    private TaskList taskList;
+    private String input;
+
+    /**
+     * Constructurs a DeadlineCommand with the specified task list and input.
+     *
+     * @param taskList Task list to add new deadline task to.
+     * @param input User input containing task description and deadline.
+     */
+    public DeadlineCommand(TaskList taskList, String input) {
+        this.taskList = taskList;
+        this.input = input;
+    }
+
+    /**
+     * Executes the command to add a new deadline task to the tasklist.
+     *
+     * @throws InvalidInputException if input format was invalid.
+     */
+    public void run() throws InvalidInputException {
+        if (!hasValidFormat(input)) {
+            throw new InvalidInputException("A deadline must have a description and due by time, no...?");
+        }
+        String[] splitInput = input.split(" ", 2);
```

Comment:
Since this is an array of Strings, the name should perhaps be `splitInputs` according to our convention